data_diff/diff_tables.py

```diff
@@ -239,12 +236,31 @@ def _diff_tables(self, table1, table2, level=0, segment_index=None, segment_coun
         for res in self._thread_map(list, diff_iters):
             yield from res
 
+    def _diff_tables(self, table1, table2, level=0, segment_index=None, segment_count=None):
+        logger.info(
+            ". " * level
+            + f"Diffing segment {segment_index}/{segment_count}, key-range: {table1.start_key:x}..{table2.end_key:x}"
+        )
+
+        (count1, checksum1), (count2, checksum2) = self._threaded_call("count_and_checksum", [table1, table2])
+
+        if not (count1 or count2):
+            logger.warn(
+                "Uneven distribution of keys detected. "
+                "For better performance, we recommend to increase the bisection-threshold."
```

Comment:
👍🏻 Good. I think it might also be worth highlighting in the warning that it will still work.

By the way, will this log to stderr automatically? If you're piping this into a file where you want all the rows that are different to go, then we need to make sure these warnings aren't printed there :)